lib/utils/job-utils/catalog-searcher.js

```diff
@@ -71,70 +69,138 @@ function searchCatalogDataFactory(
+    /**
+     * Get virtual disk catalog data
+     * @param {String} nodeId - node identifier
+     * @return {Promise} drive virtual disk catalogs
+     */
+    function getVirtualDiskCatalog(nodeId){
+        return waterline.catalogs.findMostRecent({
+            node: nodeId,
+            source: 'megaraid-virtual-disks'
         })
-        .tap(function (driveIds) {
+        .then(function (virtualDiskCatalog) {
+            if (!_.has(virtualDiskCatalog, 'data.Controllers[0]')) {
+                return Promise.reject(
+                    new Error('Could not find megaraid-virtual-disks catalog data.'));
+            }
+            return virtualDiskCatalog.data;
+        });
+    }
+
+    /**
+     * Get physical disk catalog data
+     * @param {String} nodeId - node identifier
+     * @return {Promise} drive physical disk catalogs
+     */
+    function getPhysicalDiskCatalog(nodeId){
+        return waterline.catalogs.findMostRecent({
+            node: nodeId,
+            source: 'megaraid-physical-drives'
+        })
+        .then(function(physicalDiskCatalog){
+            if (!_.has(physicalDiskCatalog, 'data')) {
+                return Promise.reject(
+                    new Error('Could not find megaraid-physical-drives catalog data.'));
+            }
+            return physicalDiskCatalog.data;
+        });
+    }
+
+    /**
+     * Get drive RAID controller vendor from catalog 
+     * @param {String} nodeId - node identifier
+     * @return {Promise} drive RAID controller vendors
+     */
+    function getRaidControllerVendor(nodeId){
+        return waterline.catalogs.findMostRecent({
+            node: nodeId,
+            source: 'megaraid-controllers'
+        })
+        .then(function (controllerCatalog) {
+            if (!_.has(controllerCatalog, 'data.Controllers')) {
+                return Promise.reject(
+                    new Error('Could not find megaraid-controllers catalog data.'));
+            }
+            var vendor,
+                controllerId,
+                controllerVendors = [];
+            var controllerDataList = _.get(controllerCatalog, 'data.Controllers');
+            _.forEach(controllerDataList, function(data){
+                vendor = _.get(data, '[Response Data][Scheduled Tasks].OEMID');
+                controllerId = _.get(data, '[Command Status][Controller]');
+                controllerVendors[controllerId] = vendor.toLowerCase();
+            });
+            return controllerVendors;
+        });
+    }
+
+    /**
+     * Get extended driveId catalog
+     * @param {String} nodeId - node identifier
+     * @param {Object} driveIds - driveId list retrieved from driveId catalogs
+     * @param {Boolean} extendJbod - flag for if we should extend JBOD physical information
+     * @return {Promise} Drive catalogs extended with Megaraid information
+     */
+    function getDriveIdCatalogExt(nodeId, filter, extendJbod) {
```

Comment:
You are right, I am struggling if we should extend disks for JBODs. For secure erase actually we don't need it. But we can leave it as it is in case we will use it in future.